docs/learn/fundamentals/fees-resource-limits-metering.mdx

```diff
@@ -0,0 +1,212 @@
+
+### Refundable and non-refundable resource fees
+
+The resource fee is calculated with a non-refundable fees portion and a refundable fees portion: `ResourceFee(sorobanData.resourceFee) = Non-refundable resource fee + Refundable resource fees`.
+
+**Non-refundable fees:** calculated from CPU instructions, read bytes, write bytes, and bandwidth (transaction size, including its signatures).
+
+**Refundable fees:** calculated from rent, events, and return value. Refundable fees are charged from the source account before the transaction is executed and then refunded based on actual usage. However, the transaction will fail if `refundableFee` is not enough to cover the actual resource usage.
+
+### Find a transaction’s resource fee
+
+The best way to find the required resource fee for any smart contract transaction is to use the [`simulateTransaction` endpoint](../smart-contract-internals/contract-interactions/transaction-simulation) from the RPC, which enables you to send a preflight transaction that will return the necessary resource values and resource fee.
+
+### Resource limitations
+
+:::note
+
+Only smart contract transactions are subject to resource limitations.
+
+:::
+
+Stellar’s ledger close time is constrained to a few seconds, preventing the execution of arbitrarily large transactions, regardless of the resource fees involved. All resources mentioned in the prior section are subject to a per-transaction limit. A transaction’s memory (RAM) is also capped, though not subject to any charge.
+
+Resource limits are determined by a validator vote and can be adjusted based on network usage and ecosystem needs with a validator consensus.
+
+Find current resource limits in the [Resource Limits & Fees](../../reference/resource-limits-fees.mdx) page in the Reference section.
+
+## Inclusion fee
+
+The inclusion fee is the maximum bid (a bid denotes a dynamic fee, meaning it varies based on certain network conditions) the submitter is willing to pay for the transaction to be included in the ledger. The inclusion fee equals the number of operations in the transaction multiplied by the effective base fee for the given ledger:
+
+inclusion fee = # of operations \* effective base fee
+
+**Effective base fee:** the fee required per operation for a transaction to make it to the ledger. This cannot be lower than 100 stroops per operation (the network minimum).
+
+**Stroop:** the smallest unit of a lumen, one ten-millionth of a lumen (.0000001 XLM).
+
+:::note
+
+Transactions can have up to 100 operations per transaction except for transactions that execute a smart contract. Smart contract transactions are only allowed one operation per transaction (unless the transaction is getting [fee-bumped](../encyclopedia/fee-bump-transactions.mdx); this would add another operation), and the limits are instead specified in CPU instructions and other limits.
```

Comment:
```suggestion
Transactions can have up to 100 operations per transaction except for transactions that execute a smart contract. Smart contract transactions are only allowed one operation per transaction (unless the transaction is getting [fee-bumped](../encyclopedia/fee-bump-transactions.mdx); this would add another operation), and the limits are instead specified in CPU instructions and other resource limits.
```